samples/star-wars-api/HttpHandlers.fs

```diff
@@ -175,68 +174,113 @@ module HttpHandlers =
-                /// Check for the conditions that would make this an introspection query
-                if isGet then
-                    logger.LogTrace ("GraphQL request is GET")
-                    return IntrospectionQuery ValueNone
+            /// Check if the request is an introspection query.
+            let isRequestIntrospection ()  = task {
+                if request.Method = HttpMethods.Get then
+                    logger.LogTrace ("Request is GET.  Must be introspection query.")
+                    return true
                 else
                     let! hasBody = checkIfHasBody()
                     if not hasBody then
-                        logger.LogTrace ("GraphQL request is not GET, request has no body")
-                        return IntrospectionQuery ValueNone
+                        logger.LogTrace ("Request is not GET but has no body.  Must be introspection query.")
+                        return true
                     else
-                        let! request = ctx.BindJsonAsync<GQLRequestContent>()
-                        if Introspection.IntrospectionQuery.Contains request.Query
-                        then
-                            logger.LogTrace ("GraphQL request is not GET, request has a body, body contains introspection query")
-                            return ValueSome request.Query |> IntrospectionQuery
-                        else
-                            logger.LogTrace ("GraphQL request is not GET, request has a body, body does not contain default introspection query")
-                            return OperationQuery request
+                        logger.LogTrace ("Request is not GET and has a body.")
+                        return false
             }
 
+            /// Check if the document is an introspection query
+            let isDocumentIntrospection (ast:Ast.Document) (operationName: string option) : bool =
+
+                let getOperation = function
+                    | OperationDefinition odef -> Some odef
+                    | _ -> None
+
+                let findOperation doc opName =
+                    match ast.Definitions |> List.choose getOperation, opName with
+                    | [def], _ -> Some def
+                    | defs, name ->
+                        defs
+                        |> List.tryFind (fun def -> def.Name = name)
+
+                match findOperation ast operationName with
+                | None ->
+                    logger.LogTrace ("Document has no operation.")
+                    false
+                | Some operation ->
+                    if not (operation.OperationType = Query) then
+                        logger.LogTrace ("Document operation is not of type Query.")
+                        false
+                    else
+                        let metaTypeFields =
+                            [ "__type"; "__schema"; "__typename" ]
+                            |> Set.ofList
```

Comment:
I usually use `System.Collections.Immutable` assuming that their performance is better. But it can be false
```suggestion
                            seq { "__type"; "__schema"; "__typename" }
                            |> Seq.toHashSet
```